### DISCUSSION TITLE
torch.nn.modules.module.ModuleAttributeError: 'StaticUnconditionalGAN' object has no attribute '_next_scale_int'

### DISCUSSION THREAD

**Discussion by aloreggia** [open-mmlab/mmgeneration#448] [Models] [answered]
Dear authors,
I'm trying to train on an UnconditionalImageDataset using stylegan2_base I got the following error:
torch.nn.modules.module.ModuleAttributeError: 'StaticUnconditionalGAN' object has no attribute '_next_scale_int'

Attached is the config file.
Best,
Andrea

[polyp.txt](https://github.com/open-mmlab/mmgeneration/files/7789542/polyp.txt)

**Comment by LeoXing1996** [comment]
Hey @aloreggia, thanks for your interest in our project.
It seems that you use `PGGANFetchDataHook` in your custom hook. However, this is should only be used for PGGAN or styleGAN1.

**Comment by LeoXing1996** [comment, marked as answer]
Besides misusing `PGGANFetchDataHook`, `scale` in `Resize` augmentation in the data pipeline should be the same as `out_size` of the `StyleGANv2Generator`. In your config, `scale` and `out_size` are 256 and 128 respectively.

**Comment by aloreggia** [comment]
Thank you for your swift reply, now it is working. Sorry, for the silly questions, I'm pretty new with GANs for data augmentation. Do you have any tips for better using your framework for this purpose?

Thanks,
Andrea

**Comment by LeoXing1996** [comment]
This is not a silly question at all 😄 . You can try to fine-tune a well-trained GAN with your data.
